Sources/SwiftDocC/Infrastructure/Workspace/DocumentationBundle+Info.swift

```diff
@@ -39,6 +49,10 @@ extension DocumentationBundle {
         /// The keys that must be present in an Info.plist file in order for doc compilation to proceed.
         static let requiredKeys: Set<CodingKeys> = [.displayName, .identifier]
         
+        /// The flag to enable or disable symbol availability inference from the module default availability.
+        ///  If not specified the default befaviout will be ``InheritDefaultAvailabilityOptions.platformAndVersion``
+        public var inheritDefaultAvailability: InheritDefaultAvailabilityOptions?
```

Comment:
I can't quite make up my mind about how I would prefer to expose this information in the public API.

Allow me to think out loud for a bit ... 🤔 

Currently, this option/configuration only determines whether or not symbols without explicit availability (either from in-source attributes or `@Available` directives) will display the versions specified in the Info.plist as the "introduced" version on the render page, or if those symbols will only display the platform name without an introduced version.

The key question to ask ourselves is: "how do we expect this to change in the future?"

If we think that we'll add some other configuration that's independent of this version-behavior, then we would basically need to create a second enum to avoid needing to add enum cases that represent combinations of behaviors:
```swift
case platformAndVersion
case platformOnly
case platformAndVersionWithNewBehavior
case platformOnlyWithNewBehavior
```

In this case, the `InheritDefaultAvailabilityOptions` won't extend beyond its original two cases. 

If both the new and the old configuration are booleans (or other two-value configuration) then we could represent them using an `OptionSet` to allow for arbitrary combinations of them.

On the other hand, if the new configuration _isn't_ a boolean, then we would need to separate it regardless. If the two configuration still are related, then maybe we would want to use a structure to group them.

All that leads me to two mini concussions:
- We possibly won't get much future extensibility from using an enum for this configuration. 
- Maybe there's a more specific name we could use for `InheritDefaultAvailabilityOptions` if it only determines this specific version behavior.

🤔

Reply:
Some of the many possibilities—names are just placeholders—that I can imagine for this are: 

### A boolean value

```swift
var displayInheritedVersionNumbers = true
```

This would make the current call site 
```swift
if !bundle.info.displayInheritedVersionNumbers { ... }
```

If we add more configuration in the future that uses either of the other two possibilities, we could convert this to a computed property that reads from the option set or struct.

We could also use a "negative" boolean, for example `var skipInheritedVersionNumbers = false` to avoid the negation at the call site.

  
### An option set

```swift
struct DefaultAvailabilityOptions: OptionSet {
    let rawValue: Int
            
    static let displayInheritedVersionNumbers = Self(rawValue: 1 << 0)
}
var defaultAvailabilityOptions: DefaultAvailabilityOptions
```

This would make the current call site 
```swift
if !bundle.info.defaultAvailabilityOptions.contains(.displayInheritedVersionNumbers) { ... }
```

If we add more _boolean_ configuration it could be defined as new values in the option set. 

New _non_-boolean configuration would need to separate properties.

### A struct with configuration as different properties

```swift
struct DefaultAvailabilityOptions {
    var displayInheritedVersionNumbers = true
}
var defaultAvailabilityOptions: DefaultAvailabilityOptions
```

This would make the current call site 
```swift
if !bundle.info.defaultAvailabilityOptions.displayInheritedVersionNumbers { ... }
```

If we add more configuration in the future they could be additional members of this struct.

----

Each of these alternatives (and other) have tradeoffs between now and the future. 

If we believe that future changes are not-that-likely then it doesn't matter all that much. 

Regardless of what we do now, we can always deprecate it and make it derive its value from the future value. The only case where that doesn't work is if we want to reuse the name but change the type (for example form an enum to a struct).

Reply:
These vague ideas are a bit hard to talk about in written form like this. Let me know if you would like to about it in person instead. 

I have no real concerns with this code. Just hoping for a brief conversation about the public API before it's harder to change.

Reply:
Same goes for [this comment](https://github.com/swiftlang/swift-docc/pull/1024/files#r1755176232) which is about the property list representation of the same configuration.